src/Data/Unfoldable1.purs

```diff
@@ -0,0 +1,72 @@
+module Data.Unfoldable1
+  ( class Unfoldable1, unfoldr1
+  , replicate1
+  , replicate1A
+  , singleton
+  , range
+  ) where
+
+import Prelude
+
+import Data.Maybe (Maybe(..))
+import Data.Semigroup.Traversable (class Traversable1, sequence1)
+import Data.Tuple (Tuple(..))
+
+-- | This class identifies non-empty data structures which can be _unfolded_.
+-- |
+-- | The generating function `f` corresponds to the `uncons` operation of a
+-- | non-empty list or array; it always return a value, and then optionally
+-- | a value to continue unfolding from.
+class Unfoldable1 t where
+  unfoldr1 :: forall a b. (b -> Tuple a (Maybe b)) -> b -> t a
+
+-- | Replicate a value `n` times. At least one value will be produced, so values
+-- | `n < 1` less than one will be ignored.
+-- |
+-- | ``` purescript
+-- | replicate1 0 "foo" == NEL.singleton "foo" :: NEL.NonEmptyList String
+-- | replicate1 2 "foo" == NEL.cons "foo" (NEL.singleton "foo") :: NEL.NonEmptyList String
+-- | ```
+replicate1 :: forall f a. Unfoldable1 f => Int -> a -> f a
+replicate1 n v = unfoldr1 step (n - 1)
+  where
+    step :: Int -> Tuple a (Maybe Int)
+    step i
+      | i <= 0 = Tuple v Nothing
+      | otherwise = Tuple v (Just (i - 1))
+
+-- | Perform an `Applicative` action `n` times (at least once, so values `n < 1`
```

Comment:
Not sure if you want this to still say `Applicative` thought "Perform an `Apply` action" doesn't read well either.